src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2697,6 +2698,9 @@ private String launchCommandIn(ArgumentListBuilder args, File workDir, EnvVars e
 
             return stdout;
         } catch (GitException | InterruptedException e) {
+            if (getHostKeyFactory() instanceof AcceptFirstConnectionVerifier) {
+                listener.getLogger().println("If you are using OpenSSH < 7.6 please choose another strategy to verify ssh host key in ‘Manage Jenkins’ -> ‘Configure Global Security’ -> ‘Git Host Key Verification Configuration'");
```

Comment:
Can we tighten the exception check for this specific message? For example, `if (e.getMessage().contains("unsupported option "accept-new")) { ... }`.

Also, it could be nice to make the message into a hyperlink to https://plugins.jenkins.io/git-client/#plugin-content-ssh-host-key-verification using `HyperlinkNote`.

Also, I think you need to update `README.md` to explain the new default setting.

One other thing - for Jenkins instances on fresh servers, `KnownHostsFileVerificationStrategy` will fail because `known_hosts` will be empty, right? That was one nice thing about `AcceptFirstConnectionVerifier`. Maybe we should also check what that error message is like and print a special message here in that case as well.

Reply:
>Can we tighten the check for printing this message? For example, if (e.getMessage().contains("unsupported option "accept-new")) { ... }.

>Also, it could be nice to make the message into a hyperlink to https://plugins.jenkins.io/git-client/#plugin-content-ssh-host-key-verification using HyperlinkNote.

>Also, I think you need to update README.md to update the reference to the default setting. Maybe a help.html file also needs to be updated?

done in [9dea3f2](https://github.com/jenkinsci/git-client-plugin/pull/882/commits/9dea3f22efd47be226dd568c9dcb866f8b99e1b1)


> One other thing - for Jenkins instances on fresh servers, KnownHostsFileVerificationStrategy will fail because known_hosts will be empty, right? That was one nice thing about AcceptFirstConnectionVerifier. Maybe we should also check what that error message is like and print a special message here in that case as well.

In that case the error message would be a 'usual' message, signalling that no host keys was provided for Host, see 
console screenshots:

Cli git:
<img width="1427" alt="CliGit console" src="https://user-images.githubusercontent.com/47667812/181995198-5d3cde36-4cab-402a-90be-72e2556d04f5.png">
 
JGit:
<img width="1192" alt="JGit console" src="https://user-images.githubusercontent.com/47667812/181995216-b44ef365-1111-4f64-825c-8e0fb849fc1e.png">

I guess on fresh servers `known_hosts` file most likely not exists, what if we manually check it and print hint message in that case? see [f8d6af0](https://github.com/jenkinsci/git-client-plugin/pull/882/commits/f8d6af0386d821e1b00b2b6105a727974752a733)